pkg/execution/executor/executor.go

```diff
@@ -745,6 +745,12 @@ func (e *executor) Execute(ctx context.Context, id state.Identifier, item queue.
 		if !errors.Is(err, state.ErrDuplicateResponse) && err != nil {
 			return nil, err
 		}
+
+		if hasPendingSteps {
+			// Other steps are pending before we re-enter the function, so
+			// we're now done with this execution.
+			return nil, nil
```

Comment:
Does this introduce `nil, nil` as a return tuple for the first time? Do we know that all callers are checking if both return values are `nil`, rather than just error?